test/string_view-test/string_view-test.cpp

```diff
@@ -0,0 +1,349 @@
+	is_true(sv_4.at(3) == str_2[3], "sv_4.at(3)");
+	is_true(sv_4.at(6) == str_2[6], "sv_4.at(6)");
+	is_true(sv_4.at(9) == str_2[9], "sv_4.at(9)");
+	is_true(sv_4.at(11) == str_2[11], "sv_4.at(11)");
+
+	is_true(sv_4.front() == str_2[0], "sv_4.front()");
+	is_true(sv_4.back() == str_2[sv_4.size() - 1], "sv_4.back()");
+
+
+	//5. Iterator support
+	//forward iterators
+	char sExpected[]{ "support" };
+	string_view sv_iter(sExpected);
+
+	bool iteratorTest{ true };
+	size_t size = sizeof(sExpected) - 1;
+	auto it = sv_iter.begin(), endIt = sv_iter.end();
+	size_t i;
+	for (i = 0; i < size && iteratorTest; ++i, ++it)
+		iteratorTest = (it != endIt && *it == sExpected[i]);
+
+	is_true(iteratorTest, "forward iterator order and content");
+	is_true(it == endIt, "forward iterator termination");
+
+	iteratorTest = true;
+	it = sv_iter.cbegin();
+	endIt = sv_iter.cend();
+	for (i = 0; i < size && iteratorTest; ++i, ++it)
+		iteratorTest = (it != endIt && *it == sExpected[i]);
+
+	is_true(iteratorTest, "forward const iterator order and content");
+	is_true(it == endIt, "forward const iterator termination");
+
+	//reverse iterators
+	char svrExpected[]{ "123456789" };
+	string_view sv_reverse("987654321");
+
+	iteratorTest = true;
+	size = sizeof(svrExpected) - 1;
+	auto rit = sv_reverse.rbegin(), rendIt = sv_reverse.rend();
+	for (i = 0; i < size && iteratorTest; ++i, ++rit)
+		iteratorTest = (rit != rendIt && *rit == svrExpected[i]);
+
+	is_true(iteratorTest, "reverse iterator order and content");
+	is_true(rit == rendIt, "reverse iterator termination");
+
+	iteratorTest = true;
+	rit = sv_reverse.crbegin();
+	rendIt = sv_reverse.crend();
+	for (i = 0; i < size && iteratorTest; ++i, ++rit)
+		iteratorTest = (rit != rendIt && *rit == svrExpected[i]);
+
+	is_true(iteratorTest, "reverse iterator order and content");
+	is_true(rit == rendIt, "reverse iterator termination");
+
+	//iterator on empty string_view: the loop body should not execute
+	iteratorTest = true;
+	for (const auto emp : empty_sv)
+		iteratorTest = false;
+	is_true(iteratorTest, "iterator on empty array");
+
+
+	//6. Modifiers
+	const char* sv_data{ sv.data() };
+	size_t sv_size{ sv.size() };
+	size_t n{ 1 };
+
+	sv.remove_prefix(n);
+	sv_size -= n;
+	is_true(sv.data() == sv_data + 1, "sv.remove_prefix().data()");
+	is_true(sv.size() == sv_size, "sv.remove_prefix().size()");
+	
+	sv_data = sv.data();
+
+	sv.remove_suffix(n);
+	sv_size -= n;
+	is_true(sv.data() == sv_data, "sv.remove_suffix().data()");
+	is_true(sv.size() == sv_size, "sv.remove_suffix().size()");
+
+	//swap(exchanges the values of *this and s.)
+	string_view o_sv("lunch!");
+	string_view s_sv("dinner");
+	char owExpected[]{ "lunch!" };
+	char swExpected[]{ "dinner" };
+	o_sv.swap(s_sv);
+	bool swapTest = true;
+	for (std::size_t idx = 0; idx < o_sv.size() && swapTest; ++idx)
+		swapTest = o_sv[idx] == swExpected[idx] && s_sv[idx] == owExpected[idx];
+	is_true(swapTest, "o_sv.swap(s_sv)");
```

Comment:
This is very hard to follow as the variable names are confusing. Is there a reason you can't just use `string_view`'s `==` operator here?

```suggestion
	//swap(exchanges the values of *this and s.)
	string_view sv_1("lunch!");
	string_view sv_2("dinner");
	char lunch[]{ "lunch!" };
	char dinner[]{ "dinner" };
	sv_1.swap(sv_2);
	is_true(sv_1 == dinner && sv_2 == lunch, "sv_1.swap(sv_2)")
```
 I understand this relies on the implemented `==` function to be correct which perhaps we shouldn't assume since it is one of the things being tested. If not using the `==` operator, perhaps write a function that compares a `string_view` to a `c-string` and returns true if the values are equal to make it more readable.

Reply:
I agree with @Ndemco. A robust test here needs to be: `sv1.data() == dinner && sv1.size() == sizeof(dinner) && sv1 == dinner;`. Of course, three more conjunctive predicates are needed against sv2 for the complete test.

Reply:
> This is very hard to follow as the variable names are confusing. Is there a reason you can't just use string_view's == operator here?

Thanks @Ndemco for the question, sorry about that my code is hard to read.

The reason I use for loop comparison here is because we can‘t just compare whether the pointers are the same. If the want sure that a `string_view` is equal to a `C-string`, then each character in this `string_view` and `C-string` must be equal.

But I think your writing is also correct, because when you use the operator `==` to compare, it also compares each character because of `string_view` overloads this operator. 

@smurthys may have a better explanation.